pkg/api/validation/validation.go

```diff
@@ -1101,6 +1115,43 @@ func validateObjectFieldSelector(fs *api.ObjectFieldSelector, expressions *sets.
 	return allErrs
 }
 
+func validateContainerResourceFieldSelector(fs *api.ResourceFieldSelector, expressions *sets.String, fldPath *field.Path, volumeDapi bool) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if volumeDapi && len(fs.ContainerName) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("containerName"), ""))
+	} else if len(fs.Resource) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("resource"), ""))
+	} else if !expressions.Has(fs.Resource) {
```

Comment:
I would just reference the constant inline instead of making it a parameter to this method.